std/assembly/string.ts

```diff
@@ -189,8 +200,71 @@ function isWhiteSpaceOrLineTerminator(c: u16): bool {
 }
 
 // @binding(CALL, [ STRING, PASS_THRU ], PASS_THRU)
-export function parseInt(str: string, radix: i32 = 10): f64 {
-  throw new Error("not implemented");
+export function parseInt(str: String, radix: i32 = 0): f64 {
+  var len = <u32>str.length;
+  if (len == 0 || (radix != 0 && radix < 2) || radix > 36)
+    return NaN;
+
+  var s0: u16 = str.charCodeAt(0);
+  var pos: u32 = 0;
+  var neg = (s0 == cn);
+
+  if (s0 == cp || neg) {
+    if (len == 1)
+      return NaN;
+
+    pos = 1;
+  }
+
+  if (radix == 0) {
```

Comment:
Wouldn't this be sufficiently covered by a `radix: i32 = 10` parameter above?

Reply:
These are fully justified changes. You can try in js:

```js
console.log(parseInt(10)) // => 10
or
console.log(parseInt('10')) // => 10
```

It's clear. But what about specified radix?
```js
console.log(parseInt('0x10', 10)) => 0
console.log(parseInt('0x10')) // => 16
```

So radix with zero value mean actually auto radix

Reply:
I see, makes sense.